Sources/AblyAssetTrackingInternal/AblyWrapper/SDK/AblySDKProtocols.swift

```diff
@@ -53,4 +57,6 @@ public protocol AblySDKAuth {
 }
 
 //sourcery: AutoMockable
-public protocol AblySDKEventListener {}
+public protocol AblySDKEventListener {
+    func underlyingListener() -> ARTEventListener
```

Comment:
I think this may come back to the discussion we had about the fact that `ably-cocoa` deals in callbacks rather than identified objects when adding listeners to the connection/channel.

So the `ARTEventListener` that's returned by cocoa is the object reference that has to be passed back to ably-cocoa to successfully "off" the listener, hence why I added a way to get at it to allow this to happen.

I'll have a think and see if I can do this without exposing the type :)

Reply:
I think that the "proper" way to do this is the same way we did it in AAT Android – [by making these `AblySDK*` types generic](https://github.com/ably/ably-asset-tracking-android/blob/e258862f739488cce0ea2a1dd92691d9b232dd12/common/src/main/java/com/ably/tracking/common/AblySdk.kt#L82-L83), parameterised by the types of listener they accept. And then internally, these generic types are responsible for [digging out the ably-java listener](https://github.com/ably/ably-asset-tracking-android/blob/e258862f739488cce0ea2a1dd92691d9b232dd12/common/src/main/java/com/ably/tracking/common/DefaultAblySdk.kt#L142-L148) and passing it through to that SDK.

Reply:
But, I mainly did that on Android because I had no alternative because the listener type was not publicly constructible (see commit message for https://github.com/ably/ably-asset-tracking-android/commit/494b607ecf66df7bdb1428bd2f4414fc33ae158d). Since `ARTEventListener` is publicly constructible, we don't have the same _necessity_, but it might still be a clean thing to do.